modules/foreman_provisioning_template.py

```diff
@@ -337,13 +346,15 @@ def main():
     state = template_dict.pop('state')
     file_name = template_dict.pop('file_name', None)
 
-    if file_name:
+    if file_name or 'template' in template_dict:
+        if file_name:
+            parsed_dict = parse_template_from_file(file_name, module)
+        else:
+            parsed_dict = parse_template(template_dict['template'], module)
+        # sanitize name from template data
+        if 'name' in parsed_dict and parsed_dict['name'] == '*':
+            module.fail_json(msg="Cannot use '*' as a template name!")
```

Comment:
Actually, this code is catching a corner case, where someone is trying to shoot himself in the knee by doing something like:
```
- name: Name a template '*' (expect this to fail)
  foreman_provisioning_template:
    <...>
    template: |
      <%#
        name: *
        <...>
      -%>
```
This is not very sensible, but it would wreck havoc if we didn't catch it.
Let's say, we put a long barrel on the nailgun. Now he can only shoot his foot.
We can write a test case about that.